server/text_generation_server/models/causal_lm.py

```diff
@@ -57,14 +57,24 @@
 PAD_SEQUENCE_TO_MULTIPLE_OF = int(os.environ.get('PAD_SEQUENCE_TO_MULTIPLE_OF', 128))
 PREFILL_BATCH_BUCKET_SIZE = int(os.environ.get('PREFILL_BATCH_BUCKET_SIZE', 4))
 CHUNK_SIZES = [1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048]
+LAZY_MODE = int(os.environ.get('PT_HPU_LAZY_MODE', 1))
 
+# Put a synchronization point if we are operating in lazy mode
+def htmark_step():
```

Comment:
when you set PT_HPU_LAZY_MODE=0, mark_step is noop. Do you see any issues with it ?

Reply:
good to know. I have reverted this change.